### PR TITLE
Change NID_* from enum to #define

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

File: wolfcrypt/src/evp.c [hunk 1]
```diff
@@ -6908,6 +6908,87 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD* type)
...
+            case CHACHA20_POLY1305_TYPE:
+                return NID_chacha20_poly1305;
+#endif
+
+#ifdef HAVE_CHACHA
+            case CHACHA20_TYPE:
+                return NID_chacha20;
+#endif
+
+            case NULL_CIPHER_TYPE :
```
Comment (haydenroche5) [inline wolfcrypt/src/evp.c:6984]: Should we gate this on HAVE_NULL_CIPHER?

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/evp.c [hunk 2]
```diff
@@ -6908,6 +6908,87 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD* type)
         return WOLFSSL_SUCCESS;
     }
 
+
+    int wolfSSL_EVP_CIPHER_CTX_nid(const WOLFSSL_EVP_CIPHER_CTX *ctx)
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply wolfcrypt/src/evp.c:6912]: Added a small test.

File: wolfcrypt/src/evp.c [hunk 3]
```diff
@@ -6908,6 +6908,87 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD* type)
...
+    {
+        WOLFSSL_ENTER("wolfSSL_EVP_CIPHER_CTX_nid");
+        if (ctx == NULL) {
+            WOLFSSL_MSG("Bad parameters");
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply wolfcrypt/src/evp.c]: Fixed.

File: wolfcrypt/src/evp.c [hunk 4]
```diff
@@ -6908,6 +6908,87 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD* type)
...
+                return NID_des_ede3_cbc;
+#endif
+#ifdef WOLFSSL_DES_ECB
+            case DES_ECB_TYPE :
+                return NID_des_ecb;
+            case DES_EDE3_ECB_TYPE :
+                return NID_des_ede3_ecb;
+#endif
+
+            case ARC4_TYPE :
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply wolfcrypt/src/evp.c:6971]: This is how this switch statement is defined in `ssl.c` too.

File: wolfcrypt/src/evp.c [hunk 5]
```diff
@@ -6908,6 +6908,87 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD* type)
...
+                return NID_chacha20_poly1305;
+#endif
+
+#ifdef HAVE_CHACHA
+            case CHACHA20_TYPE:
+                return NID_chacha20;
+#endif
+
+            case NULL_CIPHER_TYPE :
+                WOLFSSL_MSG("Null cipher has no NID");
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply wolfcrypt/src/evp.c]: Fixed.

**Review by haydenroche5 — APPROVED** [review]

**Pull request merged**